osu.Game/Skinning/LegacyBeatmapSkin.cs

```diff
@@ -59,6 +62,14 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
             return base.GetConfig<TLookup, TValue>(lookup);
         }
 
+        protected override IBindable<Color4> GetComboColour(IHasComboColours source, int comboIndex, IHasComboInformation combo)
+        {
+            if (combo is IHasLegacyBeatmapComboOffset legacyBeatmapCombo)
+                return base.GetComboColour(source, legacyBeatmapCombo.LegacyBeatmapComboIndex, combo);
```

Comment:
Just to confirm, everything else in this PR is just supporting code to make this specific conditional happen?

Reply:
Yeah, https://github.com/ppy/osu/pull/12682 refactoring the whole combo colour retrieval logic so that skins can override that and provide their own of index ("legacy beatmap combo index" in this case).

I can't really think of a valid reason as to why I've made combo offsets legacy rather than that because `LegacyBeatmapSkin` is the one using it, but it's a pointless reason because there's no other "beatmap skin" class that would've otherwise used it.

I can revert the whole `ComboOffset`s change and instead expose a new `BeatmapOffsetComboIndex` in `IHasComboInformation` or something, needs a better name though...

Reply:
I think that sounds better, but I'd need to see it to make a call. I wouldn't worry too much about naming until we are happy with the structure.

Reply:
I've pushed the changes here to make it easy to review as well.